algosdk/future/transaction.py

```diff
@@ -588,6 +622,8 @@ def __init__(
             raise error.KeyregOnlineTxnInitError("votelst")
         if votekd is None:
             raise error.KeyregOnlineTxnInitError("votekd")
+        if state_proof_ID is None:
+            raise error.KeyregOnlineTxnInitError("state_proof_ID")
```

Comment:
In order to maintain compatibility before state proof keys are required, I think this should allow the state proof key to be None.